src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -591,6 +622,101 @@ protected void deleteNetworks(Azure azureService, List<com.microsoft.azure.manag
         return executeScriptOnVM(vm, instanceScript);
     }
 
+    private Optional<VirtualMachineExtension> retrieveExistingExtension(VirtualMachine vm) {
+
+        if (vm.osType().equals(OperatingSystemTypes.LINUX)) {
+            return vm.listExtensions()
+                     .values()
+                     .stream()
+                     .filter(extension -> extension.publisherName().equals(SCRIPT_EXTENSION_PUBLISHER_LINUX) &&
+                                          extension.typeName().equals(SCRIPT_EXTENSION_TYPE_LINUX))
+                     .findAny();
+        } else if (vm.osType().equals(OperatingSystemTypes.WINDOWS)) {
+            return vm.listExtensions()
+                     .values()
+                     .stream()
+                     .filter(extension -> extension.publisherName().equals(SCRIPT_EXTENSION_PUBLISHER_WINDOWS) &&
+                                          extension.typeName().equals(SCRIPT_EXTENSION_TYPE_WINDOWS))
+                     .findAny();
+        } else {
+            throw new RuntimeException(unsupportedOperatingSystemError(vm.osType().toString()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)